### ISSUE TITLE
hosted world crashing when version updates

### ISSUE THREAD

**Issue by johnkslg** [StrangeLoopGames/EcoIssues#14712]
```
[07:29:22] Failed to load object of type 'CivicsData'. It will be ignored, but may cause inconsistency and errors in other entities: Exception has been thrown by the target of an invocation.
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at Eco.Core.Serialization.Serializers.ClassSerializer`1.OnDecodeComplete(Object instance) in C:\Jenkins\workspace\Eco Staging\Server\Eco.Core\Serialization\Serializers\ClassSerializer.cs:line 186
   at Eco.Core.Serialization.Serializers.ClassSerializer.DecodeInstance(BinaryReader reader) in C:\Jenkins\workspace\Eco Staging\Server\Eco.Core\Serialization\Serializers\ClassSerializer.cs:line 49
   at Eco.Core.Serialization.Serializers.ObjectSerializer.DecodeObject(BinaryReader reader, Int32 objTypeID) in C:\Jenkins\workspace\Eco Staging\Server\Eco.Core\Serialization\Serializers\ObjectSerializer.cs:line 151
   at Eco.Core.Serialization.Serializers.ObjectSerializer.DecodeReferenceable(BinaryReader reader) in C:\Jenkins\workspace\Eco Staging\Server\Eco.Core\Serialization\Serializers\ObjectSerializer.cs:line 120
   at Eco.Core.Serialization.Serializers.ObjectSerializer.DecodeToMember(BinaryReader reader, Object instance, MemberInfo memberInfo) in C:\Jenkins\workspace\Eco Staging\Server\Eco.Core\Serialization\Serializers\ObjectSerializer.cs:line 165
   at Eco.Core.Serialization.Serializers.ClassSerializer`1.DecodeToInstance(Object& instance, BinaryReader reader) in C:\Jenkins\workspace\Eco Staging\Server\Eco.Core\Serialization\Serializers\ClassSerializer.cs:line 177
   at Eco.Core.Serialization.Serializers.ClassSerializer.DecodeInstance(BinaryReader reader) in C:\Jenkins\workspace\Eco Staging\Server\Eco.Core\Serialization\Serializers\ClassSerializer.cs:line 49
   at Eco.Core.Serialization.Serializers.ObjectSerializer.DecodeObject(BinaryReader reader, Int32 objTypeID) in C:\Jenkins\workspace\Eco Staging\Server\Eco.Core\Serialization\Serializers\ObjectSerializer.cs:line 151
   at Eco.Core.Serialization.Serializers.ObjectSerializer.DecodeObject(BinaryReader reader, Int32 objTypeID) in C:\Jenkins\workspace\Eco Staging\Server\Eco.Core\Serialization\Serializers\ObjectSerializer.cs:line 151
   at Eco.Core.Serialization.Serializers.ObjectSerializer.DecodeReferenceable(BinaryReader reader) in C:\Jenkins\workspace\Eco Staging\Server\Eco.Core\Serialization\Serializers\ObjectSerializer.cs:line 120
   at Eco.Core.Serialization.Serializers.ObjectSerializer.Decode(BinaryReader reader) in C:\Jenkins\workspace\Eco Staging\Server\Eco.Core\Serialization\Serializers\ObjectSerializer.cs:line 129
   at Eco.Core.Serialization.SimpleSerializer.Deserialize(Stream stream) in C:\Jenkins\workspace\Eco Staging\Server\Eco.Core\Serialization\SimpleSerializer.cs:line 648
   at Eco.Core.Serialization.DataStore.DeserializeSafe[T](Stream stream) in C:\Jenkins\workspace\Eco Staging\Server\Eco.Core\Serialization\DataStore.cs:line 614
An element with the same key but a different value already exists. Key: basic election
   at System.Collections.Immutable.ImmutableDictionary`2.HashBucket.Add(TKey key, TValue value, IEqualityComparer`1 keyOnlyComparer, IEqualityComparer`1 valueComparer, KeyCollisionBehavior behavior, OperationResult& result)
   at System.Collections.Immutable.ImmutableDictionary`2.AddRange(IEnumerable`1 items, MutationInput origin, KeyCollisionBehavior collisionBehavior)
   at System.Collections.Immutable.ImmutableDictionary`2.AddRange(IEnumerable`1 pairs, Boolean avoidToHashMap)
   at System.Collections.Immutable.ImmutableDictionary`2.AddRange(IEnumerable`1 pairs)
   at Eco.Core.Utils.ThreadSafeDictionary`2.<>c__DisplayClass25_0.<AddRange>b__0(ImmutableDictionary`2 x) in C:\Jenkins\workspace\Eco Staging\Server\Eco.Core\Utils\ThreadSafeDictionary.cs:line 49
   at Eco.Core.Utils.ImmutableHelper.ApplyImmutable[T](T& original, Func`2 apply) in C:\Jenkins\workspace\Eco Staging\Server\Eco.Core\Utils\ImmutableHelper.cs:line 19
   at Eco.Core.Utils.ThreadSafeDictionary`2.AddRange(IEnumerable`1 pairs) in C:\Jenkins\workspace\Eco Staging\Server\Eco.Core\Utils\ThreadSafeDictionary.cs:line 49
   at Eco.Core.Utils.ControllerDictionary`2.<>n__2(IEnumerable`1 pairs)
   at Eco.Core.Utils.ControllerDictionary`2.<>c__DisplayClass10_0.<AddRange>b__0() in C:\Jenkins\workspace\Eco Staging\Server\Eco.Core\Utils\ControllerDictionary.cs:line 29
   at Eco.Core.Utils.ControllerDictionary`2.ApplyChange(Action change) in C:\Jenkins\workspace\Eco Staging\Server\Eco.Core\Utils\ControllerDictionary.cs:line 47
   at Eco.Core.Utils.ControllerDictionary`2.AddRange(IEnumerable`1 pairs) in C:\Jenkins\workspace\Eco Staging\Server\Eco.Core\Utils\ControllerDictionary.cs:line 29
   at Eco.Core.Systems.Registrar.OnDeserialized() in C:\Jenkins\workspace\Eco Staging\Server\Eco.Core\Systems\Registrar.cs:line 216
[07:29:22] Loading Civics... Finished in 18.4ms
[07:29:23] Starting ControllerManager...
[07:29:23] Starting ControllerManager... Finished in 0.0ms
[07:29:23] [Server Initialization]... Finished in 35,552.0ms
[07:29:23] Failed to load server, Exception was Exception: InvalidOperationException
Message:Singleton already assigned.
Source:Eco.Shared

System.InvalidOperationException: Singleton already assigned.
   at Eco.Shared.Utils.Singleton`1..ctor()
   at Eco.Gameplay.Civics.CivicsData..ctor()
```

**Comment by johnkslg** [comment]
Basic Election getting duplicated somehow when bumping version
Havent seen this on any of my builds, is it Hosted Worlds aka Linux specific?
Including @beastyn if you can isolate a repro
Including @mirasrael if you have ideas for migration side

**Comment by beastyn** [comment]
saves from HW
https://www.dropbox.com/s/jcc7jxw76360lw0/Game.eco?dl=0
https://www.dropbox.com/s/w4cz52wykoyc261/Game.db?dl=0

**Comment by johnkslg** [comment]
Fixed so these worlds load now.
Somehow there were duplicate 'Basic Election' entries.  I added detection and handling for this case:
![image](https://user-images.githubusercontent.com/3536496/74375001-6b5d5a00-4d94-11ea-8a2c-f0d815f5e064.png)

I couldnt figure out why this happened though, so we'll have to keep an eye on it to see if it keeps happening. But the game should load and work fine now even when problems like this happen.
